File: content/DP/lab2/src/main/java/ru/prokdo/model/database/operational/File.java
package ru.prokdo.model.database.operational;

import ru.prokdo.model.database.EntityType;


public final class File extends OperationalEntity {
    public File(int id, String name) {
        this.id = id;
        this.name = name;

        this.entityType = EntityType.FILE;
    }

    @Override
    public String toString() {
        return String.format("File[id=%d, name=%s]", this.id, this.name);
    }
}

File: content/DP/lab3/src/main/kotlin/model/database/entity/File.kt
package model.database.entity

import model.database.access.AccessType

class File(name: String, accessType: Int) : Entity() {
    override val entityType: EntityType
    override val id: Int
    override var name: String = ""
        set(value) {
            if (value.isEmpty() || value.isBlank()) throw IllegalArgumentException("Entity name cannot be empty or blank")

            field = value
        }
    override var accessType: Int = -1
        set(value) {
            if (!AccessType.ACCESS_RANGE.contains(value)) throw IllegalArgumentException("Unknown access type")

            field = value
        }

    init {
        if (name.isEmpty() || name.isBlank()) throw IllegalArgumentException("Entity name cannot be empty or blank")

        this.entityType = EntityType.FILE
        this.id = name.hashCode()
        this.name = name
        this.accessType = accessType
    }
}
